blazor/treeview/node-editing.md

```diff
@@ -122,4 +122,2073 @@ When editing is completed by focus out or by pressing the **Enter** key, the mod
+    .bootstrap5-dark .treeviewdiv .treeCount.e-badge {
+        background-color: #008800;
+    }
+
+    .highcontrast .treeviewdiv .treeCount.e-badge {
+        color: white;
+    }
+
+    .nodetext {
+        float: left;
+    }
+
+    .e-rtl .nodetext {
+        float: right;
+    }
+
+    .nodebadge {
+        float: right;
+        margin-right: 5px
+    }
+
+    .e-rtl .nodebadge {
+        float: left;
+        margin-left: 5px
+    }
+</style>
+
+```
+
+## Read-only TreeView
+
+The Blazor TreeView component can be set to read-only mode by disabling certain features or handling events that trigger actions, such as disabling the context menu, handling item selected, item expand, and item collapse events, and disabling the editable property of the TreeView component, which will prevent for adding, removing, or editing nodes.
+
+```cshtml
+<SfTreeView TValue="MailItem">
+    <TreeViewFieldsSettings TValue="MailItem" Id="Id" DataSource="@MyFolder" Text="FolderName" ParentID="ParentId" HasChildren="HasSubFolders" Expanded="Expanded"></TreeViewFieldsSettings>
+    <TreeViewEvents TValue="MailItem" NodeSelecting="NodeSelecting" NodeExpanding="NodeExpanding" NodeCollapsing="NodeCollapsing"></TreeViewEvents>
+</SfTreeView>
+
+@code {
+    public class MailItem
+    {
+        public string Id { get; set; }
+        public string ParentId { get; set; }
+        public string FolderName { get; set; }
+        public bool Expanded { get; set; }
+        public bool HasSubFolders { get; set; }
+    }
+    List<MailItem> MyFolder = new List<MailItem>();
+    public void NodeSelecting(NodeSelectEventArgs args)
+    {
+        args.Cancel = true;
+    }
+    public void NodeExpanding(NodeExpandEventArgs args)
+    {
+        args.Cancel = true;
+    }
+    public void NodeCollapsing(NodeExpandEventArgs args)
+    {
+        args.Cancel = true;
+    }
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+        MyFolder.Add(new MailItem
+            {
+                Id = "1",
+                FolderName = "Inbox",
+                HasSubFolders = true,
+                Expanded = true
+            });
+        MyFolder.Add(new MailItem
+            {
+                Id = "2",
+                ParentId = "1",
+                FolderName = "Categories",
+                Expanded = true,
+                HasSubFolders = true
+            });
+        MyFolder.Add(new MailItem
+            {
+                Id = "3",
+                ParentId = "2",
+                FolderName = "Primary"
+            });
+        MyFolder.Add(new MailItem
+            {
+                Id = "4",
+                ParentId = "2",
+                FolderName = "Social"
+            });
+        MyFolder.Add(new MailItem
+            {
+                Id = "5",
+                ParentId = "2",
+                FolderName = "Promotions"
+            });
+    }
+}
+
+```
+
+## Node editing deny for a particular node
```

Comment:
Change heading "Restrict Node Editing for Specific Node"

Reply:
Done.